odl/tomo/geometry/parallel.py

```diff
@@ -322,16 +553,38 @@ def rotation_matrix(self, angles):
 
     def __repr__(self):
         """Return ``repr(self)``."""
-        # TODO: det_init_axes
-        inner_fstr = '\n    {!r},\n    {!r}'
-
-        if not np.allclose(self.origin_to_det, [1, 0]):
-            inner_fstr += ',\n    det_init_pos={det_init_pos!r}'
-
-        inner_str = inner_fstr.format(self.motion_partition,
-                                      self.det_partition,
-                                      det_init_pos=self._det_init_pos)
-        return '{}({})'.format(self.__class__.__name__, inner_str)
+        posargs = [self.motion_partition, self.det_partition]
+        optargs = []
+
+        if not np.allclose(self.extra_rotation, np.eye(self.ndim)):
+            inv_rot = np.linalg.inv(self.extra_rotation)
+            orig_det_pos = inv_rot.dot(self.det_pos_init)
+            orig_det_axes = [inv_rot.dot(a) for a in self.det_axes_init]
+        else:
+            orig_det_pos = self.det_pos_init
+            orig_det_axes = self.det_axes_init
+
+        # TODO: change array printing from tolist() to new array_str
+        def_init_pos = self._default_config['det_pos_init']
+        if not np.allclose(orig_det_pos, def_init_pos):
+            optargs.append(('det_pos_init', orig_det_pos.tolist(), None))
+            if np.linalg.norm(orig_det_pos) == 0:
+                init_rot = np.eye(self.ndim)
+            else:
+                init_rot = rotation_matrix_from_to(def_init_pos, orig_det_pos)
```

Comment:
All of this would be much clearer if we simply applied the matrix when calling the `det_init_pos` etc functions, just saying :)